executor/runcexecutor/executor.go

```diff
@@ -466,10 +433,60 @@ func (s *forwardIO) Stderr() io.ReadCloser {
 type startingProcess struct {
 	Process *os.Process
 	ready   chan struct{}
+	ended   chan struct{}
+}
+
+// newStartingProcess will create a startingProcess that will be monitored, where
```

Comment:
This can be follow-up but I wonder what a better name would be for this. Is it like `runcProcessHandle`?

Reply:
Also wonder if the signature could be

```
h, ctx := newRuncProcessHandle(ctx, ...)
```

The returned context would be the one that is based on `Background()` and doesn't immediately get canceled when the input context gets canceled.